test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs

```diff
@@ -598,5 +599,112 @@ public void HelpTextShowsExtendedHelp()
 
             Assert.Contains(app.ExtendedHelpText, app.GetHelpText());
         }
+
+        [Theory]
+        [InlineData(new[] { "--version", "--flag" }, "1.0")]
+        [InlineData(new[] { "-V", "-f" }, "1.0")]
+        [InlineData(new[] { "--help", "--flag" }, "some flag")]
+        [InlineData(new[] { "-h", "-f" }, "some flag")]
+        public void HelpAndVersionOptionStopProcessing(string[] input, string expectedOutData)
+        {
+            using (StringWriter outWriter = new StringWriter())
+            {
+                var app = new CommandLineApplication { Out = outWriter };
+                app.HelpOption("-h --help");
+                app.VersionOption("-V --version", "1", "1.0");
+                var optFlag = app.Option("-f |--flag", "some flag", CommandOptionType.NoValue);
+
+                app.Execute(input);
+
+                outWriter.Flush();
+                string outData = outWriter.ToString();
+                Assert.Contains(expectedOutData, outData);
+                Assert.False(optFlag.HasValue());
+            }
+        }
+
+        [Theory]
+        [InlineData("-f:File1", "-f:File2")]
+        [InlineData("--file:File1", "--file:File2")]
+        [InlineData("--file", "File1", "--file", "File2")]
+        public void ThrowsExceptionOnSingleValueOptionHavingTwoValues(params string[] inputOptions)
+        {
+            var app = new CommandLineApplication();
+            app.Option("-f |--file", "some file", CommandOptionType.SingleValue);
+
+            var exception = Assert.Throws<CommandParsingException>(() => app.Execute(inputOptions));
+
+            Assert.Equal("Unexpected value 'File2' for option 'file'", exception.Message);
+        }
+
+        [Theory]
+        [InlineData("-v")]
+        [InlineData("--verbose")]
+        public void NoValueOptionCanBeSet(string input)
+        {
+            var app = new CommandLineApplication();
+            var optVerbose = app.Option("-v |--verbose", "be verbose", CommandOptionType.NoValue);
+
+            app.Execute(input);
+
+            Assert.True(optVerbose.HasValue());
+        }
+
+        [Fact(Skip = "As of version 1.1.0 giving the help option aborts processing without settig it")]
```

Comment:
Hmm, did this behavior change between 1.0.0 and 1.1.0? This is probably a bug we should fix.